### ISSUE TITLE
Custom analysis_options.yaml is not working if does not contain rules only

### ISSUE THREAD

**Issue by matpag** [insideapp-oss/sonar-flutter#34]
We have a custom `analysis_options.yaml` which is in this form:

```analyzer:
  exclude:
    - "**/*.dart_tool/*"
    - "**/build/*"
    - "**/*.g.dart"

  strong-mode:
    implicit-casts: false
    implicit-dynamic: false

linter:
  rules:
    # Errors
    - avoid_empty_else
    - control_flow_in_finally
    - empty_statements
    - test_types_in_equals
    - throw_in_finally
    - valid_regexps
    - camel_case_types
```

But even when using `sonar.dart.analysis.useExistingOptions=true` in the properties, our rules are ignored and SonarQube dashboard reports 0 errors.

I think the problem is that when enabling this flag, the rules are parsed from the start without searching for the "linter" string first, breaking the analyzation. 
If I remove everything before the `linter:` string, the rules are picked correctly.

CC @stephanecodes

**Comment by amatkivskiy** [comment]
I have the same question. It seems this plugin supports only built-in rules.
